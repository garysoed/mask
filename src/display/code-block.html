<style>
  .hljs {
    overflow: auto;
    background: var(--mkThemePassiveBG1B);
    color: var(--mkThemePassiveFG1B);
    padding: 1rem;
    font-family: var(--mkFontFamilyMono);
  }

  .hljs-emphasis {
    font-style: italic;
  }

  .hljs-strong {
    font-weight: var(--mkFontWeightBold);
  }

  .hljs-keyword,
  .hljs-selector-tag {
    color: var(--mkThemeHoverBG1B);
    font-weight: var(--mkFontWeightBold);
  }

  .hljs-bullet,.hljs-string,.hljs-symbol {
    color: var(--mkThemeActionFG2B);
  }

  .hljs-comment,
  .hljs-quote {
    color: var(--mkThemeDisabledFG2B);
  }

  .hljs-built_in,
  .hljs-builtin-name,
  .hljs-literal,
  .hljs-meta,
  .hljs-type {
    color: var(--mkThemeActionBG1B);
    font-weight: var(--mkFontWeightBold);
  }

  .hljs-params,
  .hljs-number {
    color: var(--mkThemePassiveFG2B);
  }

  .hljs-attribute,
  .hljs-link,
  .hljs-name,
  .hljs-selector-class,
  .hljs-selector-id,
  .hljs-tag,
  .hljs-template-variable,
  .hljs-variable {
    color: var(--mkThemeFocusBG2B);
  }

  .hljs-regexp {
    color: var(--mkThemeFocusBG2B);
  }

  .hljs-section,
  .hljs-title {
    color: var(--mkThemePassiveFG1B);
    font-weight: var(--mkFontWeightBold);
  }

  .hljs-addition,
  .hljs-deletion {
    color: #1b1818;
    display: inline-block;
    width: 100%
  }

  .hljs-deletion {
    background-color: #ca4949
  }

  .hljs-addition {
    background-color: #4b8b8b
  }
</style>
<pre fill-parent id="root"></pre>
