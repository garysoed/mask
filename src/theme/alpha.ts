export enum Alpha {
  HIGH,
  MED_HIGH,
  MED_LOW,
  LOW,
  VERY_LOW,
}

export function getAlphaNumber(alpha: Alpha): number {
  switch (alpha) {
    case Alpha.HIGH:
      return 1;
    case Alpha.MED_HIGH:
      return 0.75;
    case Alpha.MED_LOW:
      return 0.65;
    case Alpha.LOW:
      return 0.45;
    case Alpha.VERY_LOW:
      return 0.35;
  }
}
