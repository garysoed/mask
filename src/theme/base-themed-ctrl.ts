import {BaseCtrl, PersonaContext} from 'persona';
import {Observable} from 'rxjs';
import {map, pairwise, startWith, tap} from 'rxjs/operators';

import {$themeLoader, _p} from '../app/app';


@_p.baseCustomElement({})
export abstract class BaseThemedCtrl<S extends {}> extends BaseCtrl<S> {
  protected readonly themeLoader$ = $themeLoader.get(this.vine);

  constructor(context: PersonaContext, specs: S) {
    super(context, specs);

    this.addSetup(this.setupThemeUpdate());
  }

  private setupThemeUpdate(): Observable<unknown> {
    // TODO: Replace with Persona's single.
    return this.themeLoader$.pipe(
        map(themeLoader => themeLoader.createElement(this.shadowRoot.ownerDocument)),
        startWith(null),
        pairwise(),
        tap(([oldEl, newEl]) => {
          if (oldEl) {
            oldEl.remove();
          }

          if (newEl) {
            this.shadowRoot.appendChild(newEl);
          }
        }),
    );
  }
}
