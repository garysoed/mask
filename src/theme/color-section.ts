export enum ColorSection {
  PASSIVE = 'Passive',
  ACTION = 'Action',
  ACTION_FOCUS = 'ActionFocus',
  ACTION_DISABLED = 'ActionDisabled',
  ACTIVE = 'Active',
  ACTION_PRIMARY = 'ActionPrimary',
  ACTION_PRIMARY_FOCUS = 'ActionPrimaryFocus',
  ACTION_PRIMARY_DISABLED = 'ActionPrimaryDisabled',
}
