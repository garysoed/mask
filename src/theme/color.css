[mk-theme="light"] {
  /*{themeLight}*/
}

[mk-theme="dark"] {
  /*{themeDark}*/
}

[mk-theme="highlight"] {
  /*{themeHighlight}*/
}
