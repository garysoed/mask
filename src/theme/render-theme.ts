import {Context, ElementSpec, ocase, renderNode, RenderSpec, root} from 'persona';
import {RenderValueFn} from 'persona/export/internal';
import {Observable, OperatorFunction} from 'rxjs';

import {$themeLoader} from '../app/app';

import {ThemeLoader} from './loader/theme-loader';


const SELECTOR = root({
  styleEl: ocase<ThemeLoader>(),
});

export function renderTheme(
    context: Context<ElementSpec>,
    themeSelector?: (fn: RenderValueFn<ThemeLoader>) => OperatorFunction<ThemeLoader, ThemeLoader>,
): Observable<unknown> {
  const renderContext = {
    document: context.element.ownerDocument,
    vine: context.vine,
  };

  if (themeSelector) {
    return $themeLoader.get(context.vine).pipe(
        themeSelector(themeLoader => renderThemeLoader(themeLoader, context)),
    );
  }

  return $themeLoader.get(context.vine).pipe(
      SELECTOR.styleEl(context.shadowRoot, renderContext)(
          themeLoader => renderThemeLoader(themeLoader, context),
      ),
  );
}

function renderThemeLoader(themeLoader: ThemeLoader, context: Context<ElementSpec>): RenderSpec {
  return renderNode({
    node: themeLoader.createElement(context.shadowRoot.ownerDocument),
  });
}