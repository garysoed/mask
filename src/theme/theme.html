<style>
  tr > td:first-child {
    padding-right: 2rem;
    text-align: end;
  }

  #col1 {
    width: 16rem;
  }

  #col2, #col3, #col4, #col5, #col6, #col7 {
    width: 12rem;
  }

  tbody tr {
    height: 8em;
  }

  thead th {
    text-align: center;
  }

  [mk-action-1].focus {
    --mkThemeBG: var(--mkThemeFocusBG1);
    --mkThemeFG: var(--mkThemeFocusFG1);
  }

  [mk-action-2].focus {
    --mkThemeBG: var(--mkThemeFocusBG2);
    --mkThemeFG: var(--mkThemeFocusFG2);
  }

  [mk-action-1].hover {
    --mkThemeBG: var(--mkThemeHoverBG1);
    --mkThemeFG: var(--mkThemeHoverFG1);
  }

  [mk-action-2].hover {
    --mkThemeBG: var(--mkThemeHoverBG2);
    --mkThemeFG: var(--mkThemeHoverFG2);
  }

  .label {
    font-size: 1rem;
  }

  .label.dark,
  #table.dark .label.light {
    display: none;
  }

  #table.dark .label.dark,
  .label.light {
    display: block;
  }
</style>

<table id="table" mk-theme-level="0">
  <colgroup>
    <col id="col1">
    <col id="col2">
    <col id="col3">
    <col id="col4">
    <col id="col5">
    <col id="col6">
    <col id="col7">
  </colgroup>
  <thead mk-body-2>
    <tr>
      <th></th>
      <th colspan="2">Main</th>
      <th colspan="2">Side</th>
      <th colspan="2">Highlight</th>
    </tr>
  </thead>
  <tbody>
    <!-- Passive -->
    <tr id="passiveRow">
      <td mk-body-2>passive</td>
      <td mk-passive-1 mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-passive-2 mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-passive-1 mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-passive-2 mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-passive-1 mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-passive-2 mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>
    </tr>


    <!-- Action -->
    <tr id="actionRow">
      <td mk-body-2>action</td>
      <td mk-action-1 mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>
    </tr>


    <!-- Focus -->
    <tr id="focusRow">
      <td mk-body-2>action (focus)</td>
      <td mk-action-1 class="focus" mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="focus" mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 class="focus" mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="focus" mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 class="focus" mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="focus" mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>
    </tr>


    <!-- Hover -->
    <tr id="hoverRow">
      <td mk-body-2>action (hover)</td>
      <td mk-action-1 class="hover" mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="hover" mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 class="hover" mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="hover" mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 class="hover" mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 class="hover" mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>
    </tr>


    <!-- Disabled -->
    <tr id="disabledRow">
      <td mk-body-2>action (disabled)</td>
      <td mk-action-1 mk-disabled mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-disabled mk-theme-context="main">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 mk-disabled mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-disabled mk-theme-context="side">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>

      <td mk-action-1 mk-disabled mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>A</h3>
        </div>
      </td>
      <td mk-action-2 mk-disabled mk-theme-context="highlight">
        <div flex-align="center" layout="column">
          <h3 flex mk-display>a</h3>
        </div>
      </td>
    </tr>
  </tbody>
</table>
