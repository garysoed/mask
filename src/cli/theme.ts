import * as fs from 'fs';

import commandLineArgs from 'command-line-args';
import {Color, fromCssColor} from 'gs-tools/export/color';
import {EMPTY, Observable, Subject} from 'rxjs';
import {catchError, takeUntil} from 'rxjs/operators';
import {Logger, ON_LOG_$} from 'santa';
import {CliDestination} from 'santa/export/cli';

import {Palette, PALETTE} from '../theme/palette';
import {Theme} from '../theme/theme';


const LOGGER = new Logger('@hive/main');

const OPTIONS = [
  {
    name: 'base',
    type: String,
  },
  {
    name: 'acc',
    type: String,
  },
  {
    name: 'out',
    type: String,
  },
];

const options = commandLineArgs(OPTIONS, {stopAtFirstUnknown: true});

/**
 * Completes whenever running is done.
 */
function run(): Observable<unknown> {
  const base = findColor(options.base);
  const acc = findColor(options.acc);
  const outStr = options.out;
  const theme = new Theme(base, acc);

  return new Observable(subscriber => {
    fs.writeFile(outStr, theme.generateCss(), {encoding: 'utf8'}, err => {
      if (err) {
        subscriber.error(err);
      } else {
        subscriber.next();
        subscriber.complete();
      }
    });
  });
}

function findColor(colorStr: string): Color {
  const existingColor: Color|undefined = PALETTE[colorStr.toUpperCase() as keyof Palette];
  if (existingColor) {
    return existingColor;
  }

  const cssColor = fromCssColor(colorStr);
  if (!cssColor) {
    throw new Error(`Unsupported color: ${colorStr}`);
  }

  return cssColor;
}

const onDone$ = new Subject();
const consoleLog = new CliDestination(
    entry => {
      if (entry.key === 'authurl') {
        return {showKey: false, enableFormat: false};
      }
      return {showKey: false};
    },
);
ON_LOG_$
    .pipe(takeUntil(onDone$))
    .subscribe(entry => {
      consoleLog.log(entry);
    });

run()
    .pipe(
        catchError(e => {
          LOGGER.error(e);
          return EMPTY;
        }),
    )
    .subscribe({complete: () => {
      onDone$.next();
    }});
