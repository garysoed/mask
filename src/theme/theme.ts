import { $asArray, $flat, $map, $pipe } from 'gs-tools/export/collect';
import { Color } from 'gs-tools/export/color';
import { cache } from 'gs-tools/export/data';

import generalCss from './general.css';
import { PRIMARY_DARK_HIGHLIGHT_SECTION_SPECS, PRIMARY_DARK_SECTION_SPECS, PRIMARY_LIGHT_HIGHLIGHT_SECTION_SPECS, PRIMARY_LIGHT_SECTION_SPECS, SECONDARY_DARK_HIGHLIGHT_SECTION_SPECS, SECONDARY_DARK_SECTION_SPECS, SECONDARY_LIGHT_HIGHLIGHT_SECTION_SPECS, SECONDARY_LIGHT_SECTION_SPECS, SectionSpec } from './section-spec';
import { ColorWithAlpha, createColor } from './shade';


export class Theme {
  constructor(
      private readonly document: Document,
      readonly baseColor: Color,
      readonly accentColor: Color,
  ) { }

  getStyleEl(): HTMLStyleElement {
    return this.createStyleEl();
  }

  setBaseColor(color: Color): Theme {
    return new Theme(this.document, color, this.accentColor);
  }

  setHighlightColor(color: Color): Theme {
    return new Theme(this.document, this.baseColor, color);
  }

  @cache()
  private createStyleEl(): HTMLStyleElement {
    const cssDeclarations = [
      this.generateSectionThemeCss(false),
      this.generateSectionThemeCss(true),
      this.generateHighlightThemeCss(),
    ];

    const styleEl = this.document.createElement('style');
    styleEl.innerHTML = `${cssDeclarations.join('\n\n')}\n${generalCss}`;
    return styleEl;
  }

  private generateHighlightThemeCss(): string {
    // Just use the light ones. The dark ones will be the same.
    const cssVariables = $pipe(
        [...PRIMARY_LIGHT_SECTION_SPECS, ...SECONDARY_LIGHT_SECTION_SPECS],
        $map(spec => {
          const cssNormalFgVar = getSectionVariableName(spec, true);
          const cssNormalBgVar = getSectionVariableName(spec, false);

          const highlightSpec = {...spec, isHighlight: true};
          const cssHighlightFgVar = getSectionVariableName(highlightSpec, true);
          const cssHighlightBgVar = getSectionVariableName(highlightSpec, false);

          return [
            `${cssNormalBgVar}: var(${cssHighlightBgVar});`,
            `${cssNormalFgVar}: var(${cssHighlightFgVar});`,
          ];
        }),
        $flat(),
        $asArray(),
    )
    .join('\n');

    return `[mk-theme-highlight] { ${cssVariables} }`;
  }

  private generateSectionThemeCss(isDark: boolean): string {
    const normalSectionSpecs = isDark ?
        [...PRIMARY_DARK_SECTION_SPECS, ...SECONDARY_DARK_SECTION_SPECS] :
        [...PRIMARY_LIGHT_SECTION_SPECS, ...SECONDARY_LIGHT_SECTION_SPECS];
    const highlightSectionSpecs = isDark ?
        [...PRIMARY_DARK_HIGHLIGHT_SECTION_SPECS, ...SECONDARY_DARK_HIGHLIGHT_SECTION_SPECS] :
        [...PRIMARY_LIGHT_HIGHLIGHT_SECTION_SPECS, ...SECONDARY_LIGHT_HIGHLIGHT_SECTION_SPECS];

    const cssVariables = $pipe(
        [...normalSectionSpecs, ...highlightSectionSpecs],
        $map(spec => {
          const base = spec.isBase ? this.baseColor : this.accentColor;
          const cssFgVar = getSectionVariableName(spec, true);
          const cssBgVar = getSectionVariableName(spec, false);
          const cssFgColor = generateCssColor(createColor(spec.fg, base));
          const cssBgColor = generateCssColor(createColor(spec.bg, base));
          return [
            `${cssFgVar}: ${cssFgColor};`,
            `${cssBgVar}: ${cssBgColor};`,
          ];
        }),
        $flat(),
        $asArray(),
    )
    .join('\n');

    const theme = isDark ? 'dark' : 'light';
    return `[mk-theme="${theme}"] { ${cssVariables} }`;
  }
}

function getSectionVariableName(sectionSpec: SectionSpec, isForeground: boolean): string {
  return [
    '--mkTheme',
    sectionSpec.section,
    isForeground ? 'FG' : 'BG',
    sectionSpec.isSecondary ? '2' : '1',
    sectionSpec.isHighlight ? 'H' : 'B',
  ].join('');
}

function generateCssColor({color, alpha}: ColorWithAlpha): string {
  return `rgba(${color.red},${color.green},${color.blue},${alpha})`;
}
