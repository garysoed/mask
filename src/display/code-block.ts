import {cache} from 'gs-tools/export/data';
import {Context, Ctrl, query, itarget, itext, otext, PRE, registerCustomElement} from 'persona';
import {combineLatest, Observable} from 'rxjs';
import {tap} from 'rxjs/operators';

import {renderTheme} from '../theme/render-theme';

import template from './code-block.html';


interface HighlightJs {
  highlightBlock(node: Node): void;
}

declare const hljs: HighlightJs;

export const $codeBlock = {
  host: {
    text: itext(),
  },
  shadow: {
    root: query('#root', PRE, {
      target: itarget(),
      text: otext(),
    }),
  },
};

class CodeBlock implements Ctrl {
  constructor(private readonly $: Context<typeof $codeBlock>) { }

  @cache()
  get runs(): ReadonlyArray<Observable<unknown>> {
    return [
      renderTheme(this.$),
      this.$.host.text.pipe(this.$.shadow.root.text()),
      combineLatest([
        this.$.host.text,
        this.$.shadow.root.target,
      ])
          .pipe(
              tap(([, rootTarget]) => {
                hljs.highlightBlock(rootTarget);
              }),
          ),
    ];
  }
}

export const CODE_BLOCK = registerCustomElement({
  ctrl: CodeBlock,
  spec: $codeBlock,
  tag: 'mk-code-block',
  template,
});