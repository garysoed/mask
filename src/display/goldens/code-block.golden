{
  "type": "element",
  "raw": "<mk-code-block mk-theme=\"light\">\n      import {something} from './something.ts';\n\n      something.callFn(123);\n      <gst-fake-shadow><style>\n  .hljs {\n    overflow: auto;\n    background: var(--mkThemePassiveBG1B);\n    color: var(--mkThemePassiveFG1B);\n    padding: 1rem;\n    font-family: var(--mkFontFamilyMono);\n  }\n\n  .hljs-emphasis {\n    font-style: italic;\n  }\n\n  .hljs-strong {\n    font-weight: var(--mkFontWeightBold);\n  }\n\n  .hljs-keyword,\n  .hljs-selector-tag {\n    color: var(--mkThemeHoverBG1B);\n    font-weight: var(--mkFontWeightBold);\n  }\n\n  .hljs-bullet,.hljs-string,.hljs-symbol {\n    color: var(--mkThemeActionFG2B);\n  }\n\n  .hljs-comment,\n  .hljs-quote {\n    color: var(--mkThemeDisabledFG2B);\n  }\n\n  .hljs-built_in,\n  .hljs-builtin-name,\n  .hljs-literal,\n  .hljs-meta,\n  .hljs-type {\n    color: var(--mkThemeActionBG1B);\n    font-weight: var(--mkFontWeightBold);\n  }\n\n  .hljs-params,\n  .hljs-number {\n    color: var(--mkThemePassiveFG2B);\n  }\n\n  .hljs-attribute,\n  .hljs-link,\n  .hljs-name,\n  .hljs-selector-class,\n  .hljs-selector-id,\n  .hljs-tag,\n  .hljs-template-variable,\n  .hljs-variable {\n    color: var(--mkThemeFocusBG2B);\n  }\n\n  .hljs-regexp {\n    color: var(--mkThemeFocusBG2B);\n  }\n\n  .hljs-section,\n  .hljs-title {\n    color: var(--mkThemePassiveFG1B);\n    font-weight: var(--mkFontWeightBold);\n  }\n\n  .hljs-addition,\n  .hljs-deletion {\n    color: #1b1818;\n    display: inline-block;\n    width: 100%\n  }\n\n  .hljs-deletion {\n    background-color: #ca4949\n  }\n\n  .hljs-addition {\n    background-color: #4b8b8b\n  }\n</style>\n<pre fill-parent=\"\" id=\"root\">\n      import {something} from './something.ts';\n\n      something.callFn(123);\n      </pre>\n<link rel=\"stylesheet\" href=\"base/test.css\"></gst-fake-shadow></mk-code-block>",
  "name": "code-block"
}